src/model/ImageDocument.cpp

```diff
@@ -16,25 +18,104 @@ std::shared_ptr<Image> createDefaultImage()
 
 QUrl defaultUrl() { return QUrl("file:////HDRV"); }
 
-ImageDocument::ImageDocument(std::shared_ptr<Image> image, QUrl const& url, QObject * parent)
+ImageDocument::ImageDocument(QUrl const& url, QObject * parent)
   : QObject(parent)
   , name_(QFileInfo(url.fileName()).baseName())
   , url_(url)
+  , activeTasks_(0)
   , scale_(1.0f)
   , brightness_(0.0f)
   , gamma_(2.2f)
-  , image_(std::move(image))
-{}
+  , image_(createDefaultImage())
+  , errorText_({ "", "", "" })
+  , watcher_(nullptr)
+  , comparisonWatcher_(nullptr)
+{
+  watcher_ = new QFutureWatcher<LoadResult>(this);
+  connect(watcher_, SIGNAL(started()), this, SIGNAL(busyChanged()));
+  connect(watcher_, SIGNAL(finished()), this, SIGNAL(busyChanged()));
+
+  auto* fsWatcher = new QFileSystemWatcher(QStringList(url_.toLocalFile()), this);
+  auto* delay = new QTimer(this);
+  delay->setSingleShot(true);
 
-ImageDocument::ImageDocument(std::shared_ptr<Image> base, std::shared_ptr<Image> comparison,
-  QUrl const& url, QObject * parent)
-  : ImageDocument(std::move(base), url, parent)
+  connect(
+    watcher_, &QFutureWatcher<LoadResult>::finished,
+    std::bind(&ImageDocument::loadFinished, this, watcher_, url_, false));
+  connect(
+    fsWatcher, &QFileSystemWatcher::fileChanged,
+    [delay]() { delay->start(500); });
+  connect(
+    delay, &QTimer::timeout,
+    std::bind(&ImageDocument::load, this, url_.toLocalFile(), watcher_));
+
+  load(url_.toLocalFile(), watcher_);
+}
+
+ImageDocument::ImageDocument(QUrl const& base, QUrl const& comparison, QObject * parent)
```

Comment:
This seems like a lot of copy paste, can we
. forward to a common constructor (or write the default values into the header)
. do something like `setupWatcher(watcher_, url_)` and `setupWatcher(comparisonWatcher_, comparisonUrl_)`